examples/05_tool_inference_lock/ex_tool_inference_lock/main.rb

```diff
@@ -0,0 +1,233 @@
+# Copyright 2021 Trimble Inc
+# Licensed under the MIT license
+
+require 'sketchup.rb'
+require 'ex_tool_inference_lock/inference_lock_helper'
+
+module Examples
+  module ToolInferenceLock
+
+    class LineTool
+
+      # Threshold in logical screen pixels for when the mouse is considered to
+      # be dragged.
+      DRAG_THRESHOLD = 10
+
+      def activate
+        @mouse_ip = Sketchup::InputPoint.new
+        @picked_first_ip = Sketchup::InputPoint.new
+        @mouse_down = ORIGIN
+        
+        # Create a InferenceLockHelper object to use in this tool.
+        # See inference_lock_helper.rb for details.
+        @inference_lock_helper = InferenceLockHelper.new(
+            self, @mouse_ip, @picked_first_ip)
+        
+        update_ui
+      end
+
+      def deactivate(view)
+        view.invalidate
+      end
+
+      def enableVCB?
+        picked_first_point?
+      end
+
+      def resume(view)
+        update_ui
+        view.invalidate
+      end
+
+      def suspend(view)
+        view.invalidate
+      end
+
+      def onCancel(reason, view)
+        reset_tool
+        view.invalidate
+      end
+      
+      def onKeyDown(key, _repeat, _flags, view)
+        # Let InferenceLockHelper object know a key was pressed.
```

Comment:
Don't think the comment adds anything the code doesn't already say.

Reply:
This is from when I tried to copy the style of existing tutorials rather than typical examples. Removing it now.